### PR TITLE
Remove step_n methods and simplify transaction object creation

### PR DESCRIPTION
**Pull request by dexX7**
- cleanup of (now) unused headers and using directives

- all step_n parsing methods are removed and replaced

- instead of hijacking the core logic methods for data population, `CMPMetaDEx` and `CMPOffer` objects can be created from `CMPTransaction`

### CODE REVIEW

File: src/omnicore/mdex.cpp [hunk 1]
```diff
@@ -679,7 +666,7 @@ bool mastercore::MetaDEx_isOpen(const uint256& txid, uint32_t propertyIdForSale)
 // to save doing a second levelDB iteration if already done in calling function, pass in optional totalSold & totalBought
 std::string mastercore::MetaDEx_getStatus(const uint256& txid, uint32_t propertyIdForSale, int64_t amountForSale, int64_t totalSold, int64_t totalReceived)
 {
-    if (totalSold == -1 || totalReceived == -1) {
+    if (totalSold == -1 || totalReceived == -1) { // TODO: do we need this?
```
Comment (zathras-crypto) [inline src/omnicore/mdex.cpp]: Yeah, it's a bypass - basically we need the values for ```totalSold``` & ```totalReceived``` but they may already be available to us by being passed in.  The default values are -1 so if they're still at defaults we know we haven't been passed the necessary info and must go and fetch it via ```getMatchingTrades```, if they're not -1 though we know we've already been passed the info and can skip this step.

Hope that makes sense :)

Comment (dexX7) [inline src/omnicore/mdex.cpp]: The only use I saw was one, where the values were provided, so I wondered. :)

Comment (zathras-crypto) [inline src/omnicore/mdex.cpp]: Ahh, you may be right :)

**Comment by zathras-crypto** [comment]
This is fine to merge by the way :)

**Comment by dexX7** [comment]
> This is fine to merge by the way :)

Thanks for the very fast review. :)

**Pull request merged**